pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -88,23 +101,34 @@ func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry,
 		}
 	}
 
-	converter.ToInternal(ext, oc)
-	if err := staticValidator.Static(ext, nil, f.env.Location(), f.env.Domain(), f.env.FeatureIsSet(env.FeatureRequireD2sV3Workers), resourceID); err != nil {
-		return api.ValidationResult{
-			Status: api.ValidationStatusFailed,
-			Error: &api.ManagementErrorWithDetails{
-				Message: to.StringPtr(err.Error()),
-			},
+	if isCreate {
+		log.Print("preflight static create validation")
+		if err = staticValidator.Static(ext, nil, f.env.Location(), f.env.Domain(), f.env.FeatureIsSet(env.FeatureRequireD2sV3Workers), resourceID); err != nil {
+			return api.ValidationResult{
+				Status: api.ValidationStatusFailed,
+				Error: &api.ManagementErrorWithDetails{
+					Message: to.StringPtr(err.Error()),
+				},
+			}
 		}
-	}
-
-	if err := f.validateInstallVersion(ctx, oc); err != nil {
-		return api.ValidationResult{
-			Status: api.ValidationStatusFailed,
-			Error: &api.ManagementErrorWithDetails{
-				Code:    to.StringPtr(api.CloudErrorCodeInvalidParameter),
-				Message: to.StringPtr(err.Error()),
-			},
+		if err = f.validateInstallVersion(ctx, oc); err != nil {
```

Comment:
bug? In the original code, this `oc` has been `converter.ToInternal`'d - are we now using the External version, and that works for all API versions?

Reply:
@carvalhe let's add a test case that would have failed here and what brendan said is correct, it should be done using internal data model

Reply:
Added a version test case, and correctly use the converter.ToInternal as well

Reply:
I'm still unsure about this, to my eye, all test cases rely on the `apiVersion` static value, so everything still appears to be hitting `apiVersion := "2022-04-01"` in the tests, or maybe I'm missing the apiversion case you mentioned? In the original code, we called `toInternal` before `staticValidator.Static`, which is what is catching my attention here. TBH, I'm not entirely sure if that is required or not, just noting it's different/standing out to me that we've changed order of calls. If we have confirmation that this code works as expected in multiple RP API versions, then I'm fine with the change but I don't see that in the tests atm.

Reply:
+@SudoBrendan 's comments.  It may not have any impact actually, we'd have to examine which fields are being dropped on external conversion and if we're performing static validation on them to figure out if there's an actual issue with the representation of what we're statically validating.  

I'd prefer consistency with `openshiftcluster_putorpatch.go` though.

Reply:
So the current test I have is for an incorrect apiversion which gets a failed case we desire. I tried other valid apiversions that seemed to work fine. I understand wanting the consistency with putorpath so I can move the toInternal call before static validate gets called (at least for create as update doesn't call toInternal before static.)